### PR TITLE
NRF91: more work towards a usable port

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- nRF91: add mcuboot support
- arch/nrf91: modem shmem always at the RAM start
- arch/nrf91: modem depends on LFCLK
- arch/nrf91: add errata workarounds
- arch/nrf91/spu: various fixes for SPU
- arch/nrf91/nrf91_modem_os.c: use vsyslog instead of syslog
- arch/nrf91/clock: various fixes for LFCLK
- nrf91/common: add configuration checks for linker script
- arch/nrf91: add POWER definitions
- nrf9160-dk: add timer support
- armv8-m/arm_securefault.c: fix warning
- nrf91/common: add support for non-secure boot
- nrf9160-dk add secure miniboot configuration that jumpt to non-secure code
- nrf9160-dk add non-secure modem configuration

The next step is to implement the socket interface integration.

## Impact
We can boot from a secure bootlaoder based on NuttX miniboot to a non-secure application that successfully initializes nrf91 modem.

## Testing
nrf9160dk

### CODE REVIEW

File: arch/arm/src/nrf91/nrf91_flash.c [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_progmem_pagesize
+ *
+ * Description:
+ *   Return page size
+ *
```
Comment (acassis) [inline arch/arm/src/nrf91/nrf91_flash.c:76]: Please include the input "size_t page" parameter

File: arch/arm/src/nrf91/nrf91_flash.c [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+      return nrf91_get_page_size();
+    }
+}
+
+/****************************************************************************
+ * Name: up_progmem_erasesize
+ *
+ * Description:
+ *   Return erase block size
+ *
```
Comment (acassis) [inline arch/arm/src/nrf91/nrf91_flash.c:96]: Ditto: size_t block

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/nrf91/nrf91_flash.c (hunk 2 above)
Comment (raiden00pl) [reply arch/arm/src/nrf91/nrf91_flash.c:96]: I'll fix these in the next PR

File: arch/arm/src/nrf91/nrf91_errata.c [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+   */
+
+  putreg32(REGULATORS_DCDCEN_ENABLE, NRF91_REGULATORS_DCDCEN);
+#endif
+
+#ifdef NRF91_LFXO_ERRATA31_WORKAROUND
+  /* Workaround for 3.15 [31] LFXO: LFXO startup fails */
+
+  *((volatile uint32_t *)0x5000470c) = 0x0;
+  *((volatile uint32_t *)0x50004710) = 0x1;
```
Comment (pkarashchenko) [inline arch/arm/src/nrf91/nrf91_errata.c:161]: Why not `putreg32`?

Comment (raiden00pl) [reply arch/arm/src/nrf91/nrf91_errata.c:161]: To keep it compatible with the errata from Nordic. They use this format to define a workaround.

Comment (pkarashchenko) [reply arch/arm/src/nrf91/nrf91_errata.c:161]: Thanks for explanation.

**Pull request merged**